### PR TITLE
[framework] use ImageFilter for zoom page transition

### LINKED ISSUE #100972: Improve the memory footprint of the zoom page transition
On https://github.com/flutter/flutter/commit/543a7e858828e721258b8fb9ded62d7a2ff05808, several flutter_gallery benchmarks are showing a 10-20MB regression in memory footprint.

https://flutter-flutter-perf.skia.org/e/?begin=1648252513&end=1648446610&keys=X845c9ddf84d6ac62deff51c6cee6ad22&num_commits=50&request_type=1&xbaroffset=28090

Weirdly, SkiaPerf flags the following commit, but it's pretty clear from the chart where the regression happened.

@dnfield @AlexV525

### PR DESCRIPTION
**Pull request by jonahwilliams**
Potentially fixes https://github.com/flutter/flutter/issues/100972 though we need full devicelab metrics.

### CODE REVIEW

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/page_transitions_theme.dart [hunk 1]
```diff
@@ -325,7 +325,11 @@ class _ZoomEnterTransition extends StatelessWidget {
       },
       child: FadeTransition(
         opacity: fadeTransition,
-        child: ScaleTransition(scale: scaleTransition, child: child),
+        child: ScaleTransition(
+          scale: scaleTransition,
+          filterQuality: FilterQuality.none,
```
Comment (dnfield) [inline packages/flutter/lib/src/material/page_transitions_theme.dart:330]: without knowing the background this seems strange. Should we add comments? Or is the idea that the docs on ScaleTransition cover this?

Comment (jonahwilliams) [reply packages/flutter/lib/src/material/page_transitions_theme.dart:330]: What seems strange?

Comment (jonahwilliams) [reply packages/flutter/lib/src/material/page_transitions_theme.dart:330]: Its documented on those widgets that it uses an ImageFilter if one is provided, though the documentation isn't extensive: https://api.flutter.dev/flutter/widgets/ScaleTransition/filterQuality.html

Comment (flar) [reply packages/flutter/lib/src/material/page_transitions_theme.dart:330]: I would have chosen FilterQuality.low. I don't think there will be much performance difference, but low will be smoother.

**Comment by jonahwilliams** [comment]
This was reverted due to regressing numerous devicelab performance metrics. To reland, we'd likely need to wait until we have better cooperation among the opacity/image filter layers raster caching approach

**Comment by flar** [comment]
Yes, I saw that in my own prototype testing so I'm not surprised. Do we have an Issue filed for looking at the interaction between those 2 layers? I imagine it might be similar to the problems recently spotted with opacity interacting poorly with its own DL child: https://github.com/flutter/flutter/issues/101872

**Pull request merged**